### ISSUE TITLE
Custom filtering rules ignored with DNS allowlist

### ISSUE THREAD

**Issue by eggplanting** [AdguardTeam/AdGuardHome#2371]
With the latest version of AdGuard Home (0.104.3) custom filtering rules get ignored, if a matching DNS allowlist entry exists.

Please add an option to choose if custom filter rules or DNS list entries are prefered.

**Example:**
Custom filterung rule: ||google.com^$client='192.168.0.82'
DNS allowlist: https://raw.githubusercontent.com/notracking/hosts-blocklists-scripts/master/hostnames.whitelist.txt

**Result:**
<img width="1177" alt="google-whitelisted" src="https://user-images.githubusercontent.com/57031979/100543049-31d7db80-324e-11eb-9e09-bf6836da3b61.png">

**Comment by ainar-g** [comment]
<p>Thank&#xA0;you for&#xA0;your report, we'll see what&#xA0;we can do
without&#xA0;breaking other people's settings.&#x2003;An&#xA0;option is one
option (heh), but&#xA0;perhaps a&#xA0;more flexible model is&#xA0;needed.</p>

**Comment by kiwilad-nz** [comment]
I am running into this issue as well as that's actually how I would of expected this to work.

Whitelists are really good and convenient to override blacklists, but in my view custom rules should always trump both (override any backlist and/or whitelist).

**Comment by angrynerd** [comment]
yes, also running into this issue. would have thought custom is prefered.

**Comment by bkuhns** [comment]
Just ran into this. I subscribe to a generic whitelist for known services (was having trouble with Google services without it). But, that whitelist allows youtube.com, which I want to block. My custom filter to block youtube.com is being ignored because the whitelist I subscribe to contains an allow for it. Custom filters should trump both block/allow lists!

**Comment by dd900** [comment]
Bumping this issue. I'm not sure why this hasn't been properly addressed yet. Custom Rules should take priority over all other rules. If not this, then maybe add some sort of priority modifier that goes beyond the $important modifier.
